plugins/qtosgrave/osglodlabel.cpp

```diff
@@ -0,0 +1,100 @@
+#include "osglodlabel.h"
+#include "qtosg.h"
+#include <iostream>
+#include <osg/Depth>
+
+namespace qtosgrave {
+
+// OSG text label that scales by camera distance and also disappears if far away enough
+OSGLODLabel::OSGLODLabel(const std::string& label) : osg::LOD() {
+    /* Transform structure of an OSGLODLabel: 
+    *
+    * [Target Transform (usually the global transform)]
+    *           |
+    *           |
+    * [Label Offset Transform (optional)]
+    *           |
+    *           |
+    * [Global LOD (this + controls label visibility)]
+    *           |
+    *           |
+    *     [Label Geode]
+    *           |
+    *           |
+    *     [Label Text]
+    */
+
+    // Set up text element and its properties
+    osg::ref_ptr<osgText::Text> text = new osgText::Text();
+    text->setText(label);
+    text->setCharacterSize(0.05);
+    text->setAutoRotateToScreen(true);
+    text->setFont( "/usr/share/fonts/truetype/msttcorefonts/Arial.ttf" );
```

Comment:
1. Let's use a free font
2. Let's ship a free font together with openrave, like Noto-sans should be able to redistribute under its license
3. Pack it into rcc (Check Cmake)
4. read the font using osgText::readFontStream